### PR TITLE
[rpc] Parse Struct tag for RPC API endpoints

### PR DESCRIPTION
**Pull request by 666lcz**
Fixed https://github.com/MystenLabs/sui/issues/6120 by normalizing user input

### CODE REVIEW

File: crates/sui-core/src/authority.rs [hunk 1]
```diff
@@ -2272,8 +2273,9 @@ impl AuthorityState {
                     .await?
             }
             EventQuery::MoveEvent(struct_name) => {
+                let normalized_struct_name = parse_sui_struct_tag(&struct_name)?.to_string();
```
Comment (666lcz) [inline crates/sui-core/src/authority.rs:2276]: This line fixes https://github.com/MystenLabs/sui/issues/6120

File: crates/sui-json-rpc-types/src/lib.rs [hunk 2]
```diff
@@ -2160,6 +2160,7 @@ pub enum SuiEvent {
     /// Move-specific event
     #[serde(rename_all = "camelCase")]
     MoveEvent {
+        // TODO: What's the best way to serialize this using `AccountAddress::short_str_lossless` ??
         package_id: ObjectID,
```
Comment (666lcz) [inline crates/sui-json-rpc-types/src/lib.rs:2164]: cc @patrickkuo do you know what's the best way to trim leading zeros for `ObjectID` when serializing this to JSON?

Comment (666lcz) [reply crates/sui-json-rpc-types/src/lib.rs:2164]: Will handle this in a separate PR

**Review by patrickkuo — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**